### DISCUSSION TITLE
How to debounce?

### DISCUSSION THREAD

**Discussion by Liwoj** [dm4t2/vue-currency-input#325] [Q&A] [answered]
Hi!

Great library! Really enjoying using it 

However I'm struggling on implementing debounce. I'm using it with Quasar `QInput` which has a `debounce` prop so I want something similar for a wrapper around `QInput` which is using `useCurrencyInput`. As I understand it the library is attaching itself directly to underlying `input` element and is also handling firing `update:modelValue` for the wrapper so there is nothing I can do without some kind of new option (`onModelUpdate` callback) which would be called by `vue-currency-input` instead of firing `update:modelValue` for me. Right ?

As it is now I see two options:
1. Handle and debounce `update:modelValue` "outside" everywhere I'm using the wrapper - don't like it 
2. Wrap the wrapper and handle debounce on 2nd level ....meh

I would prefer to have it build-in into my wrapper (and not wrapping it twice)

Thanks!

**Comment by dm4t2** [comment, marked as answer]
Hi, thanks for your feedback. I opened a new issue targeting a more controlled emit behavior: #326. 
This will allow you to add a debounce implementation directly in your component without any additional wrappers.
